### PR TITLE
Implement coord and structure functions

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

File: src/julia_interface.jl [hunk 1]
```diff
@@ -108,6 +108,39 @@ function NLPModels.grad!(nlp :: CUTEstModel, x :: AbstractVector, g :: AbstractV
...
+
+  - nlp:  [IN] CUTEstModel
+  - x:    [IN] Array{Float64, 1}
+  - c:    [OUT] Array{Float64, 1}
+  - jrow: [OUT] Array{Int32, 1}
+  - jcol: [OUT] Array{Int32, 1}
+  - jval: [OUT] Array{Float64, 1}
+"""
+function cons_coord!(nlp :: CUTEstModel, x :: AbstractVector, c :: AbstractVector, rows :: AbstractVector{<:Integer}, cols :: AbstractVector{<:Integer}, vals :: AbstractVector)
+  nvar = nlp.meta.nvar;
```
Comment (dpo) [inline src/julia_interface.jl]: We don't need those semicolons. Eventually, we should remove them all (in PRs that only do that). They're not necessary and clutter up the code.

File: src/julia_interface.jl [hunk 2]
```diff
@@ -108,6 +108,39 @@ function NLPModels.grad!(nlp :: CUTEstModel, x :: AbstractVector, g :: AbstractV
...
+    cons_coord!(nlp, x, c, rows, cols, vals)
+
+  - nlp:  [IN] CUTEstModel
+  - x:    [IN] Array{Float64, 1}
+  - c:    [OUT] Array{Float64, 1}
+  - jrow: [OUT] Array{Int32, 1}
+  - jcol: [OUT] Array{Int32, 1}
+  - jval: [OUT] Array{Float64, 1}
+"""
+function cons_coord!(nlp :: CUTEstModel, x :: AbstractVector, c :: AbstractVector, rows :: AbstractVector{<:Integer}, cols :: AbstractVector{<:Integer}, vals :: AbstractVector)
```
Comment (dpo) [inline src/julia_interface.jl]: I don't think we want to allow other `Integer` types here. CUTEst only accepts `Cint`.

File: src/julia_interface.jl (hunk 1 above)
Comment (abelsiqueira) [reply src/julia_interface.jl]: #183

Comment (dpo) [reply src/julia_interface.jl]: Right, but let's no commit new code with semicolons.

File: src/julia_interface.jl (hunk 2 above)
Comment (abelsiqueira) [reply src/julia_interface.jl]: I've changed to use `Int32`, and convert other `Integer` to `Int32`.

**Comment by dpo** [comment]
Great, thank you!

**Comment by abelsiqueira** [comment]
Now updated, removed the `;`, changed `Integer` functions, and updated the `docs` to pass the tests. Should be ready to go.

**Pull request merged**